actors/evm/src/interpreter/execution.rs

```diff
@@ -51,49 +50,197 @@ pub struct Machine<'r, 'a, RT: Runtime + 'a> {
     pub output: Output,
 }
 
-type Instruction<M> = unsafe fn(*mut M) -> Result<(), StatusCode>;
-
 macro_rules! def_opcodes {
-    ($($op:ident)*) => {
-        def_ins_raw! {
-            UNDEFINED(_m) {
-                Err(StatusCode::UndefinedInstruction)
-            }
-        }
-        $(def_ins_raw! {
-            $op (m) {
-                instructions::$op(m)
+    ($($code:literal: $op:ident,)*) => {
+        pub const fn jumptable<'r, 'a, RT: Runtime>() -> [Instruction<'r, 'a, RT>; 256] {
+            def_ins_raw! {
+                UNDEFINED(_m) {
+                    Err(StatusCode::UndefinedInstruction)
+                }
             }
-        })*
-
-        def_jmptable! {
-            $($op)*
-        }
-    }
-}
+            $(def_ins_raw! {
+                $op (m) {
+                    instructions::$op(m)
+                }
+            })*
 
-macro_rules! def_jmptable {
-    ($($op:ident)*) => {
-        const fn jmptable() -> [Instruction<Machine<'r, 'a, RT>>; 256] {
-            let mut table: [Instruction<Machine::<'r, 'a, RT>>; 256] = [Machine::<'r, 'a, RT>::UNDEFINED; 256];
-            $(table[OpCode::$op as usize] = Machine::<'r, 'a, RT>::$op;)*
+            let mut table: [Instruction<'r, 'a, RT>; 256] = [UNDEFINED; 256];
+            $(table[$code] = $op;)*
             table
         }
+        $(pub const $op: u8 = $code;)*
     }
 }
 
 macro_rules! def_ins_raw {
     ($ins:ident ($arg:ident) $body:block) => {
         #[allow(non_snake_case)]
-        unsafe fn $ins(p: *mut Self) -> Result<(), StatusCode> {
+        unsafe fn $ins<'r, 'a, RT: Runtime>(p: *mut Machine<'r, 'a, RT>) -> Result<(), StatusCode> {
             // SAFETY: macro ensures that mut pointer is taken directly from a mutable borrow, used
             // once, then goes out of scope immediately after
-            let $arg: &mut Self = &mut *p;
+            let $arg: &mut Machine<'r, 'a, RT> = &mut *p;
             $body
         }
     };
 }
 
+pub mod opcodes {
+    use super::instructions;
+    use super::Machine;
+    use crate::interpreter::StatusCode;
+    use fil_actors_runtime::runtime::Runtime;
+
+    pub type Instruction<'r, 'a, RT> =
+        unsafe fn(*mut Machine<'r, 'a, RT>) -> Result<(), StatusCode>;
+
+    def_opcodes! {
```

Comment:
this kills the comments we had

Reply:
Those were left-over from when this table actually did different things for different instructions. Now it's _purely_ a dispatch table.